### PR TITLE
Polish the Azure.ResourceManager.ResourceHealth for first stable

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs [hunk 1]
```diff
@@ -1,56 +1,49 @@
...
+    public partial class ResourceHealthAvailabilityStatusData : Azure.ResourceManager.Models.ResourceData
     {
-        internal AvailabilityStatusData() { }
+        internal ResourceHealthAvailabilityStatusData() { }
         public Azure.Core.AzureLocation? Location { get { throw null; } }
-        public Azure.ResourceManager.ResourceHealth.Models.AvailabilityStatusProperties Properties { get { throw null; } }
+        public Azure.ResourceManager.ResourceHealth.Models.ResourceHealthAvailabilityStatusProperties Properties { get { throw null; } }
     }
-    public partial class AvailabilityStatusResource : Azure.ResourceManager.ArmResource
+    public partial class ResourceHealthAvailabilityStatusResource : Azure.ResourceManager.ArmResource
```
Comment (ArcturusZhang) [inline sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs:9]: I personally think this and ResourceHealthChildAvailabilityStatusResource do not make much sense as resources. Please consider making these non-resources.
Its name suggests it is fetch some kind of status, this does not look like a resource

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs [hunk 2]
```diff
@@ -1,56 +1,49 @@
...
-        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.ResourceHealth.Models.StatusActiveEvent> StatusActiveEvents { get { throw null; } }
-        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.ResourceHealth.Models.StatusBanner> StatusBanners { get { throw null; } }
+        public virtual Azure.Response<Azure.ResourceManager.ResourceHealth.ResourceHealthAvailabilityStatusResource> Get(string filter = null, string expand = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.ResourceHealth.ResourceHealthAvailabilityStatusResource>> GetAsync(string filter = null, string expand = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
-    public partial class EmergingIssuesGetResultResource : Azure.ResourceManager.ArmResource
+    public partial class ResourceHealthChildAvailabilityStatusResource : Azure.ResourceManager.ArmResource
```
Comment (ArcturusZhang) [inline sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs:19]: same here

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs [hunk 3]
```diff
@@ -1,56 +1,49 @@
...
+    public partial class ResourceHealthEventData : Azure.ResourceManager.Models.ResourceData
     {
-        internal EventData() { }
+        internal ResourceHealthEventData() { }
         public string AdditionalInformationMessage { get { throw null; } }
-        public string ArticleContent { get { throw null; } }
+        public Azure.ResourceManager.ResourceHealth.Models.EventArticle Article { get { throw null; } }
         public string Description { get { throw null; } }
         public int? Duration { get { throw null; } }
         public bool? EnableChatWithUs { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs:49]: should this be `IsChatWithUsEnabled`?

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs [hunk 4]
```diff
@@ -247,35 +245,32 @@ public partial class TenantResourceHealthEventImpactedResourceCollection : Azure
...
-        public static Azure.ResourceManager.ResourceHealth.Models.StatusBanner StatusBanner(string title = null, string message = null, string cloud = null, System.DateTimeOffset? lastModifiedOn = default(System.DateTimeOffset?)) { throw null; }
-        public static Azure.ResourceManager.ResourceHealth.Models.Update Update(string summary = null, System.DateTimeOffset? updateOn = default(System.DateTimeOffset?)) { throw null; }
+    public partial class AvailabilityStateRecentlyResolved
```
Comment (ArcturusZhang) [inline sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs:268]: Please consider prepend the RP name, this name is a bit generic

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs [hunk 5]
```diff
@@ -334,6 +321,44 @@ public partial class EmergingIssueImpact
...
+        public System.BinaryData Parameters { get { throw null; } }
+    }
+    public partial class EventFaq
+    {
+        internal EventFaq() { }
+        public string Answer { get { throw null; } }
+        public string LocaleCode { get { throw null; } }
+        public string Question { get { throw null; } }
+    }
+    public partial class EventImpact
```
Comment (ArcturusZhang) [inline sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs:338]: The name of these three are too generic

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs [hunk 6]
```diff
@@ -425,18 +521,11 @@ public partial class EventPropertiesRecommendedActionsItem
...
     }
-    public partial class Faq
-    {
-        internal Faq() { }
-        public string Answer { get { throw null; } }
-        public string LocaleCode { get { throw null; } }
-        public string Question { get { throw null; } }
-    }
-    public partial class Impact
+    public partial class EventUpdate
```
Comment (ArcturusZhang) [inline sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs:524]: well these models that start with `Event` look like to have a generic name, please consider to change them

Comment (ArthurMa1978) [reply sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs]: fixed

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs (hunk 5 above)
Comment (ArthurMa1978) [reply sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs]: fixed

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs (hunk 4 above)
Comment (ArthurMa1978) [reply sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs]: fixed

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs]: fixed

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs]: fixed

File: sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/resourcehealth/Azure.ResourceManager.ResourceHealth/api/Azure.ResourceManager.ResourceHealth.netstandard2.0.cs]: fixed

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**